storage/aptosdb/src/pruner/state_pruner_manager.rs

```diff
@@ -68,6 +70,19 @@ impl PrunerManager for StatePrunerManager {
         self.pruner.as_ref().min_readable_version()
     }
 
+    fn get_min_viable_version(&self) -> Version {
+        let min_version = self.get_min_readable_version();
+        if self.is_pruner_enabled() {
+            let window_cutoff_with_buffer = self.latest_version.lock().saturating_sub(
+                self.prune_window
+                    .saturating_sub(self.user_pruning_window_offset),
+            );
+            std::cmp::max(min_version, window_cutoff_with_buffer)
```

Comment:
should really just do `min(latest_version, min_version + offset)`? because the window can change between restarts

Reply:
It needs to be calculated from the right side.

Say you have versions 0-1000, and the window is 1,000,000.  Offset is (20,000)

Your calculation above will do the following:

min(1,000, 0+2,0000) -> Which will only return 1 version (1000).

Let's do the same calculation with this here (though I think there's a typo)

(1,000 - (1,000,000 + 20,000)) saturating at 0

max(0, 0) = 0

Reply:
But max(500, 1000 - (1000 - 200)) (in case the window changed from 500 to 1000) will give you 500, which makes the offset noop.  Returning only one version doesn't make querying that version fail. but returning the min_readable_version will.

Reply:
```
if !pruner_enabled || latest_version - min_readable_version > prune_window - offset {
    return min_readable_version
}
```

something like this will avoid returning only 1 version

Reply:
change to max() and I'm good